src/pydvl/influence/array.py

```diff
@@ -0,0 +1,390 @@
+
+    def aggregate_from_generator(
+        self, tensor_generator: Generator[TensorType, None, None]
+    ):
+        """
+        Aggregates tensors from a single-level generator into a list. This method simply collects
+        each tensor emitted by the generator into a single list.
+
+        Args:
+            tensor_generator: A generator that yields TensorType objects.
+
+        Returns:
+            A list containing all the tensors provided by the tensor_generator.
+        """
+        return [t for t in tensor_generator]
+
+
+class LazyArrayProvider(ABC):
+    """
+    Abstract base class representing a provider of lazily evaluated, array-like data. This class
+    is tailored for situations where data is not immediately computed or loaded, but is instead
+    generated or processed on-demand. The 'array-like' aspect refers to the data structure or
+    format rather than the ability to perform typical array operations like indexing.
+    """
+
+    @abstractmethod
+    def compute(self, tensor_aggregator: Optional[TensorAggregator] = None):
+        """
+        Triggers the computation or retrieval of the data, optionally aggregating the results using
+        a specified TensorAggregator. This method is responsible for initiating the actual data
+        processing or fetching tasks.
+
+        Args:
+            tensor_aggregator: An optional aggregator that specifies
+                how the results should be combined or processed.
+
+        Returns:
+            The result of the computation or data retrieval, processed as specified by the
+            tensor_aggregator, if provided. The exact return type and format depend on the
+            implementation and the aggregator used.
+        """
+
+    @abstractmethod
+    def to_zarr(
+        self,
+        path_or_url: str,
+        numpy_converter: NumpyConverter,
+        return_stored: bool = False,
+        overwrite: bool = False,
+    ):
+        """
+        Exports the computed or retrieved data to Zarr format, suitable for efficient storage
+        and access, especially in the context of large-scale datasets. This method allows saving
+        the data to a local file system or a cloud storage service.
+
+        Args:
+            path_or_url: The file path or URL where the Zarr-formatted data will be stored.
+            numpy_converter: A tool to convert data into numpy arrays, which are compatible with Zarr storage.
+            return_stored: If True, returns the stored Zarr data object. Defaults to False.
+            overwrite: If True, overwrites any existing data at the specified location.
+
+
+        Returns:
+            Depending on the value of return_stored, this method may return a reference to the stored
+            Zarr data object or None.
+        """
+
+
+class OneDimChunkedLazyArrayProvider(LazyArrayProvider):
+    """
+    A class representing chunked, and lazily evaluated array, where the chunking is restricted to the first dimension
```

Comment:
```suggestion
    A class representing a chunked, and lazily evaluated array, where the chunking is restricted to the first dimension
```

Also, please remember the column limit.